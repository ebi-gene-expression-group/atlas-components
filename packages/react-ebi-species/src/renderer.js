"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
