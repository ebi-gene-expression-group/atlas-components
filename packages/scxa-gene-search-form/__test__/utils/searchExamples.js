const searchExamples = [
  {
    text: `Rick Sanchez`,
    url: `http://rickandmorty.wikia.com/wiki/Special:Search?query=Rick+Sanchez`
  },
  {
    text: `Unity`,
    url: `http://rickandmorty.wikia.com/wiki/Special:Search?query=Unity`
  },
  {
    text: `Galactic Federation`,
    url: `http://rickandmorty.wikia.com/wiki/Special:Search?query=Galactic+Federation`
  },
  {
    text: `Blemflarck`,
    url: `http://rickandmorty.wikia.com/wiki/Special:Search?query=Blemflarck`
  },
  {
    text: `anatomy`,
    url: `http://rickandmorty.wikia.com/wiki/Special:Search?query=anatomy`
  }
]

export default searchExamples
