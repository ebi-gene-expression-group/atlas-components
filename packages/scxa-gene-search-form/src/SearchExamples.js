import React from 'react'
import PropTypes from 'prop-types'

const SearchExamples = ({links}) =>
  <p style={{fontSize: `smaller`}}>Examples:
    {
      links.map(
        (link, index) =>
          <span key={index}>
            {index === 0 && ` `}<a href={link.url}>{link.text}</a>{index < links.length - 1 && `, `}
          </span>)
    }
  </p>

SearchExamples.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
}

export default SearchExamples
