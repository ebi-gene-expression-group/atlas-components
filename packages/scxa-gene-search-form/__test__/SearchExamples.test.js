import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import SearchExamples from '../src/SearchExamples'

import searchExamples from './utils/searchExamples'

Enzyme.configure({ adapter: new Adapter() })

const props = {
  links: searchExamples.filter(() => Math.random() > 0.5)
}

describe(`SearchExamples`, () => {
  test(`renders one link per element`, () => {
    const wrapper = shallow(<SearchExamples {...props} />)
    expect(wrapper.find(`a`)).toHaveLength(props.links.length)
  })

  test(`matches snapshot`, () => {
    const tree =
      renderer
        .create(<SearchExamples links={searchExamples} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
