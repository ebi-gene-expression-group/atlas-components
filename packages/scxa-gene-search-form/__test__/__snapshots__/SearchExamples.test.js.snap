// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchExamples matches snapshot 1`] = `
<p
  style={
    Object {
      "fontSize": "smaller",
    }
  }
>
  Examples:
  <span>
     
    <a
      href="http://rickandmorty.wikia.com/wiki/Special:Search?query=Rick+Sanchez"
    >
      Rick Sanchez
    </a>
    , 
  </span>
  <span>
    <a
      href="http://rickandmorty.wikia.com/wiki/Special:Search?query=Unity"
    >
      Unity
    </a>
    , 
  </span>
  <span>
    <a
      href="http://rickandmorty.wikia.com/wiki/Special:Search?query=Galactic+Federation"
    >
      Galactic Federation
    </a>
    , 
  </span>
  <span>
    <a
      href="http://rickandmorty.wikia.com/wiki/Special:Search?query=Blemflarck"
    >
      Blemflarck
    </a>
    , 
  </span>
  <span>
    <a
      href="http://rickandmorty.wikia.com/wiki/Special:Search?query=anatomy"
    >
      anatomy
    </a>
  </span>
</p>
`;
